crates/sui-transaction-checks/src/lib.rs

```diff
@@ -186,13 +186,9 @@ mod checked {
         // Overrides the gas objects in the transaction.
         gas_override: &[ObjectRef],
     ) -> SuiResult<SuiGasStatus> {
-        // Cheap validity checks that is ok to run multiple times during processing.
-        transaction.check_version_and_features_supported(protocol_config)?;
```

Comment:
We can remove these checks because it has already been checked (multiple times) in the callsites.